profiler/src/ProfilerEngine/Datadog.Profiler.Native/BclEventsParser.cpp

```diff
@@ -0,0 +1,467 @@
+    if (!EventsParserHelper::Read(statusCode, pEventData, cbEventData, offset))
+    {
+        return;
+    }
+
+    _pNetworkListener->OnRequestHeaderStop(timestamp, pActivityId, statusCode);
+}
+
+void BclEventsParser::OnResponseContentStart(std::chrono::nanoseconds timestamp, LPCGUID pActivityId, LPCGUID pRelatedActivityId, LPCBYTE pEventData, ULONG cbEventData)
+{
+}
+
+void BclEventsParser::OnResponseContentStop(std::chrono::nanoseconds timestamp, LPCGUID pActivityId, LPCGUID pRelatedActivityId, LPCBYTE pEventData, ULONG cbEventData)
+{
+    _pNetworkListener->OnResponseContentStop(timestamp, pActivityId);
+}
+
+void BclEventsParser::OnRedirect(std::chrono::nanoseconds timestamp, LPCGUID pActivityId, LPCGUID pRelatedActivityId, LPCBYTE pEventData, ULONG cbEventData)
+{
+    // string redirectUrl
+
+    ULONG offset = 0;
+    WCHAR* redirectUrl = EventsParserHelper::ReadWideString(pEventData, cbEventData, &offset);
+
+    if (redirectUrl == nullptr)
+    {
+        return;
+    }
+
+    _pNetworkListener->OnRedirect(timestamp, pActivityId, shared::ToString(shared::WSTRING(redirectUrl)));
+}
+
+
+void BclEventsParser::ParseSocketsEvent(
+    std::chrono::nanoseconds timestamp,
+    DWORD version,
+    INT64 keywords,
+    DWORD id,
+    LPCBYTE pEventData,
+    ULONG cbEventData,
+    LPCGUID pActivityId,
+    LPCGUID pRelatedActivityId
+)
+{
+    switch (id)
+    {
+        case 1: // ConnectStart
+            OnConnectStart(timestamp, pActivityId, pRelatedActivityId);
+            break;
+        case 2: // ConnectStop
+            OnConnectStop(timestamp, pActivityId, pRelatedActivityId);
+            break;
+        case 3: // ConnectFailed
+            OnConnectFailed(timestamp, pActivityId, pRelatedActivityId, pEventData, cbEventData);
+            break;
+
+        // the following events are related to incoming requests that we don't monitor
+        //
+        case 4: // AcceptStart
+            break;
+        case 5: // AcceptStop
+            break;
+        case 6: // AcceptFailed
+            break;
+        default:
+            break;
+    }
+}
+
+void BclEventsParser::OnConnectStart(std::chrono::nanoseconds timestamp, LPCGUID pActivityId, LPCGUID pRelatedActivityId)
+{
+    _pNetworkListener->OnConnectStart(timestamp, pActivityId);
+}
+
+void BclEventsParser::OnConnectStop(std::chrono::nanoseconds timestamp, LPCGUID pActivityId, LPCGUID pRelatedActivityId)
+{
+    _pNetworkListener->OnConnectStop(timestamp, pActivityId);
+}
+
+void BclEventsParser::OnConnectFailed(std::chrono::nanoseconds timestamp, LPCGUID pActivityId, LPCGUID pRelatedActivityId, LPCBYTE pEventData, ULONG cbEventData)
+{
+    // string exception message
+
+    ULONG offset = 0;
+    WCHAR* message = EventsParserHelper::ReadWideString(pEventData, cbEventData, &offset);
+    if (message == nullptr)
+    {
+        return;
+    }
+
+    _pNetworkListener->OnConnectFailed(timestamp, pActivityId, shared::ToString(shared::WSTRING(message)));
```

Comment:
why no using the overload `shared::ToString(WCHAR*, std::size_t)`  instead ?
it would avoid a temporary allocation (`shared::WSTRING(message))`)